File: API/Placeful.Api/Models/Memory.cs
using Placeful.Api.Models;

namespace Placeful.Models;

public class Memory
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public Location Location { get; set; } = new Location();
    public string ImageUrl { get; set; } = string.Empty;
}

File: Frontend/placeful/lib/core/domain/models/memory.dart
import 'package:json_annotation/json_annotation.dart';
import 'location.dart';

part 'memory.g.dart';

@JsonSerializable(explicitToJson: true)
class Memory {
  final String id;
  final String title;
  final String description;
  final Location location;
  final String imageUrl;

  Memory({
    required this.id,
    this.title = '',
    this.description = '',
    required this.location,
    this.imageUrl = '',
  });

  factory Memory.fromJson(Map<String, dynamic> json) =>
      _$MemoryFromJson(json);

  Map<String, dynamic> toJson() => _$MemoryToJson(this);
}
